external-crates/move/crates/move-compiler/src/linters/almost_swapped.rs

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+//! Implements a linter check in Rust to detect unnecessary variable swap sequences in Move language code.
+//! The linter identifies consecutive assignments that effectively swap two variables without using a temporary variable.
+use crate::{
+    diag,
+    diagnostics::{
+        codes::{custom, DiagnosticInfo, Severity},
+        WarningFilters,
+    },
+    expansion::ast::ModuleIdent,
+    naming::ast::Var_,
+    parser::ast::FunctionName,
+    shared::{program_info::TypingProgramInfo, CompilationEnv},
+    typing::{
+        ast::{self as T, LValue_, SequenceItem_, UnannotatedExp_},
+        visitor::{TypingVisitorConstructor, TypingVisitorContext},
+    },
+};
+use move_ir_types::location::Loc;
+use std::collections::VecDeque;
+
+use super::{LinterDiagCategory, LINT_WARNING_PREFIX, SWAP_SEQUENCE_DIAG_CODE};
+
+const SWAP_SEQUENCE_OVERFLOW_DIAG: DiagnosticInfo = custom(
+    LINT_WARNING_PREFIX,
+    Severity::Warning,
+    LinterDiagCategory::Correctness as u8,
+    SWAP_SEQUENCE_DIAG_CODE,
+    "Unnecessary swap sequence detected. Consider simplifying the code or using a temporary variable if swapping is intended.",
+);
+
+pub struct SwapSequence;
+
+pub struct Context<'a> {
+    env: &'a mut CompilationEnv,
+    last_assignment: VecDeque<(Var_, Var_)>,
```

Comment:
Can we accomplish this with an option instead?

Reply:
No we can't